src/main/scala/uk/gov/nationalarchives/tdr/api/db/DbConnection.scala

```diff
@@ -1,7 +1,57 @@
 package uk.gov.nationalarchives.tdr.api.db
 
-import akka.stream.alpakka.slick.javadsl.SlickSession
+import akka.stream.alpakka.slick.scaladsl.SlickSession
+import com.typesafe.config.ConfigFactory
+import scalacache.CacheConfig
+import scalacache.caffeine.CaffeineCache
+import scalacache.memoization._
+import scalacache.modes.try_._
+import slick.jdbc.JdbcBackend
+import slick.jdbc.hikaricp.HikariCPJdbcDataSource
+import software.amazon.awssdk.auth.credentials.DefaultCredentialsProvider
+import software.amazon.awssdk.regions.Region
+import software.amazon.awssdk.services.rds.RdsUtilities
+import software.amazon.awssdk.services.rds.model.GenerateAuthenticationTokenRequest
+
+import scala.concurrent.duration._
+import scala.util.{Failure, Success, Try}
 
 object DbConnection {
-  val db = SlickSession.forConfig("consignmentapi").db
+  implicit val passwordCache: CaffeineCache[String] = CaffeineCache[String](CacheConfig())
+  val slickSession: SlickSession = SlickSession.forConfig("consignmentapi")
+
+  def db: JdbcBackend#DatabaseDef = {
+    val db = slickSession.db
+    db.source match {
+      case hikariDataSource: HikariCPJdbcDataSource =>
+        val configBean = hikariDataSource.ds.getHikariConfigMXBean
+        getPassword match {
+          case Failure(exception) => throw exception
+          case Success(password) =>
+            configBean.setPassword(password)
+            db
+        }
+      case _ =>
+        db
+    }
+  }
+
+  def getPassword: Try[String] = memoize[Try, String](Some(5.minutes)) {
```

Comment:
Sorry, just one more suggestion for a comment: could mention that this is chosen to be less than the 15 minute expiry and link to a doc about temporary IAM passwords?

Reply:
I've added a comment in.